.github/instructions/powershell-scripts.instructions.md

```diff
@@ -0,0 +1,138 @@
+---
+applyTo: "**/*.ps1"
+---
+
+# PowerShell Scripts (*.ps1)
+
+When writing PowerShell script files in the Azure PowerShell repository, follow these compatibility and portability guidelines:
+
+## PowerShell Version Compatibility
+
+### Default Requirement: PowerShell 5.1 Compatibility
+- **All PowerShell scripts must be compatible with PowerShell 5.1** unless they are specifically for repository tooling
+- Use syntax and cmdlets that work in both Windows PowerShell 5.1 and PowerShell 7+
+- Avoid PowerShell 7+ exclusive features in user-facing scripts
+
+### Repository Tooling Exception
+- Scripts in `/tools/`, `.azure-pipelines/`, and build/test automation may target PowerShell 7+
+- CI/CD scripts, build scripts, and development tools can use modern PowerShell features
+- Clearly document if a tool script requires PowerShell 7+
+
+## Cross-Platform Compatibility
+
+### Unix-like OS Support
+- **All scripts must be compatible with Unix-like operating systems** (Linux, macOS)
+- Avoid Windows-specific paths, registry operations, and Windows-only cmdlets
+- Use cross-platform PowerShell features and patterns
+
+### Path Handling
+```powershell
+# Good - Cross-platform path handling
+$configPath = Join-Path -Path $PSScriptRoot -ChildPath "config.json"
+$modulePath = [System.IO.Path]::Combine($PSScriptRoot, "modules", "Az.Example")
+
+# Avoid - Windows-specific paths
+$configPath = "$PSScriptRoot\config.json"
+$modulePath = "$env:USERPROFILE\Documents\PowerShell\Modules"
+```
+
+### File System Operations
+```powershell
+# Good - Cross-platform file operations
+if (Test-Path $filePath) {
+    $content = Get-Content -Path $filePath -Raw
+}
+
+# Avoid - Windows-specific approaches
+if ($env:OS -eq "Windows_NT") { 
+    # Windows-only logic
+}
```

Comment:
It's OK to have platform-specific logic if that's the business requirement. Just make sure to have graceful fallback on other platforms.